plugins/src/main/kotlin/org/gradle/kotlin/dsl/codeartifact.kt

```diff
@@ -0,0 +1,15 @@
+package org.gradle.kotlin.dsl
+
+import co.bound.codeartifact.CodeArtifactPluginExtension
+import org.gradle.api.Action
+import org.gradle.api.initialization.Settings
+
+/**
+ * Configure Maven repository for AWS CodeArtifact.
+ *
+ * @param action an [Action] to configure the [CodeArtifactPluginExtension]
+ */
+@Suppress("unused")
+fun Settings.codeartifact(action: Action<CodeArtifactPluginExtension>) {
+    extensions.configure(CodeArtifactPluginExtension::class.java, action)
+}
```

Comment:
Ah, nice!

I wasn't aware of this trick.

An alternative could have been to document the usage for Kotlin DSL as
```kotlin
configure<co.bound.codeartifact.CodeArtifactPluginExtension> {
...
}
```

But this is even better. Thanks!